Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -615,6 +615,59 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Push and Pull Attachments SG", "[.
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Push & Pull Deletion SG", "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    const string docID = idPrefix + "ppd-doc1";
+    constexpr size_t collectionCount = 1;
+    string revID;
+
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs;
+    std::array<C4Collection *, collectionCount> collections;
+    std::array<unordered_map<alloc_slice, unsigned>, collectionCount> docIDs;
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+
+    collectionSpecs = {
+        Roses
+    };
+    collections = collectionPreamble(collectionSpecs, "sguser", "password");
+    docIDs[0] = getDocIDs(collections[0]);
+    replCollections = {
+        C4ReplicationCollection{collectionSpecs[0], kC4OneShot, kC4Disabled},
+    };
+
+    C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+    
+    createRev(collections[0], slice(docID), kRevID, kFleeceBody);
+    createRev(collections[0], slice(docID), kRev2ID, kEmptyFleeceBody, kRevDeleted);
+    replicate(paramsSetter);
+
+    C4Log("-------- Deleting and re-creating database --------");
+    deleteAndRecreateDB();
+
+    collections = collectionPreamble(collectionSpecs, "sguser", "password");
+    replCollections = {
+        C4ReplicationCollection{collectionSpecs[0], kC4Disabled, kC4OneShot},
+    };
+    paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+
+    createRev(collections[0], slice(docID), kRevID, kFleeceBody);
```

Comment:
if you use verifyDocs, you don't have to recreate the 1-rev doc here.

Reply:
what is verifyDB ?

Reply:
The idea of the test is to: create a doc with 2 revs, 1st normal and 2nd deleted. Push this doc to SGW. Clear local DB. Create doc again but only with 1st (not deleted) rev, then pull into local again and make sure that the local doc was updated with the deleted rev.

Reply:
is using verifyDocs preffered?